### PR TITLE
Merge 1.4 into master

### PR DESCRIPTION
**Pull request by phaseinducer**

### CODE REVIEW

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php [hunk 1]
```diff
@@ -20,12 +25,16 @@ class ObjectDetacher implements ObjectDetacherInterface, BulkObjectDetacherInter
     /** @var ManagerRegistry */
     protected $managerRegistry;
 
+    /** @var array */
+    protected $scheduledForDirtyCheck;
```
Comment (skeleton) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: break static analysis (more than 20 characters) :(

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php [hunk 2]
```diff
@@ -48,6 +64,93 @@ public function detachAll(array $objects)
     }
 
     /**
+     * Detach entity living in the scheduledForDirtyCheck's
+     * unit of work property.
+     *
+     * @param mixed $entity  The entity to be detached
+     * @param array $visited Array of already detached entities
+     */
+    protected function doDetachScheduled($entity, array &$visited)
```
Comment (skeleton) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: should not be in a dedicated bundle ?

File: src/Akeneo/Component/Analytics/ChainedDataCollectorInterface.php [hunk 3]
```diff
@@ -8,6 +8,8 @@
  * @author    Nicolas Dupont <nicolas@akeneo.com>
  * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @deprecated Will be removed in 1.5, please use directly Akeneo\Component\Analytics\ChainedDataCollector
```
Comment (skeleton) [inline src/Akeneo/Component/Analytics/ChainedDataCollectorInterface.php]: to remove ?

Comment (phaseinducer) [inline src/Akeneo/Component/Analytics/ChainedDataCollectorInterface.php]: Will be done in another PR

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php (hunk 2 above)
Comment (phaseinducer) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: I don't understand. Ping @SamirBoulil ?

Comment (SamirBoulil) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: @skeleton I agree it is not an optimum solution.

Instead of creating a bundle to listen for doctrine's event and *reset* the problematic array. We chose to use the same technique used in mongodb (see below in the file) and manually cascade detach objects in the array in the ObjectDetacher class.

To put the fix here allows us to precisely remove objects to be detached instead of resetting the whole array like we once did in the separate bundle.

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php (hunk 1 above)
Comment (phaseinducer) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: To fix in another PR

**Comment by pierallard** [comment]
GTM

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php (hunk 2 above)
Comment (jjanvier) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: as already commented in the first PR, we should not do as is in master imvho
we should put this stuff in a dedicated "FixDoctrineBugDetacher" to isolate the problem

Comment (skeleton) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/Common/Detacher/ObjectDetacher.php]: agree with jj
you can merge like that, but we should create a card to create a dedicated bundle

**Comment by skeleton** [comment]
gtm

**Comment by nidup** [comment]
GTM, as discussed previously some re-work to do, imho, we should first merge this one, then re-work
